core/Archive.php

```diff
@@ -910,15 +920,19 @@ private function prepareArchive(array $archiveGroups, Site $site, Period $period
                 $periodDateStr,
                 $this->params->getSegment()->getOriginalString(),
                 $plugin,
-                $requestedReport
+                empty($requestedReport) && $shouldOnlyProcessRequestedArchives ? $archiveNames : $requestedReport
```

Comment:
for easier reading of the code without the need to think about how this evaluates
```suggestion
                (empty($requestedReport) && $shouldOnlyProcessRequestedArchives) ? $archiveNames : $requestedReport
```